pystac/extensions/eo.py

```diff
@@ -343,12 +363,30 @@ def cloud_cover(self) -> Optional[float]:
 
     @cloud_cover.setter
     def cloud_cover(self, v: Optional[float]) -> None:
-        self._set_property(CLOUD_COVER_PROP, v)
+        self._set_property(CLOUD_COVER_PROP, validated_percentage(v), pop_if_none=True)
```

Comment:
Added `pop_if_none` given this note in the ext "If not available the field should not be provided".